src/graphs/BasicGraphs.jl

```diff
@@ -159,6 +159,32 @@ function add_edges!(g::HasGraph, srcs::AbstractVector{Int},
   add_parts!(g, :E, n, (src=srcs, tgt=tgts, kw...))
 end
 
+""" Add reflexive edges to a graph
+
+Setting the keyword `idempotent::Bool=true` will skip adding reflexive edges to vertices which already have them
+"""
+function add_reflexives!(g::HasGraph; idempotent::Bool=false)
+  if idempotent
+    out = Int[]
+    for v in vertices(g)
+      es = incident(g, v, :src) ∩ incident(g, v, :tgt)
+      if isempty(es)
+        e′ = add_edge!(g, v, v)
+        push!(out, e′)
+      end
+      if length(es) > 1
+        error("Vertex $v has more than one reflexive loop. Set `idempotent=false` to bypass this error by allowing multiple reflexive edges")
```

Comment:
Hm the issue isn't that there are multiple reflexive edges (there can only be one in a reflexive graph) but that it gets picked arbitrarily by `convert`. But the error message is here rather than in `convert` so you'd have to phrase it differently. Maybe just say "multiple loops" rather than multiple reflexive edges?